### ISSUE TITLE
Encryption not possible with FEC?

### ISSUE THREAD

**Issue by alatteri** [CESNET/UltraGrid#16]
With FEC, seems not possible to also have Encryption.

```
root@streamsender01:~# uv -l auto -t decklink -c libavcodec:codec=H.264:bitrate=25M -f rs:200:220 --encryption password123 45.50.167.55
UltraGrid 1.5 (master rev 6981cefc built Feb 23 2019 01:19:14)

Display device   : none
Capture device   : decklink
Audio capture    : none
Audio playback   : none
MTU              : 1500 B
Video compression: libavcodec:codec=H.264:bitrate=25M
Audio codec      : PCM
Network protocol : UltraGrid RTP
Audio FEC        : none
Video FEC        : rs:200:220

Created new RTP session with SSRC 0x0c7ed290.
Display initialized-none
[DeckLink] Auto-choosen device 0.
Using device DeckLink Mini Recorder 4K
Enable video input: NTSC
[DeckLink] Trying to autodetect format.
Start capture
DeckLink capture device enabled
Video capture initialized-decklink
Created new RTP session with SSRC 0x708f3ef5.
[control] Fec changed successfully
Frame received (#0) - No input signal detected
[Decklink capture] Format change detected (display mode, color space).
[Decklink capture] Using codec: RGBA
Enable video input: 1080p23.98 DCI 2K
[lavc] Using codec: H.264, encoder: libx264
[lavc] Setting bitrate to 25000000 bps.
[lavc] Setting preset to ultrafast.
[lavc] Selected pixfmt: yuv420p
FEC symbol size: 1329, symbols per packet: 1, payload size: 1329
[Decklink capture] 113 frames in 5.03848 seconds = 22.4274 FPS
```

```
Ganymede:MacOS alanlatteri$ ./uv -d sdl --encryption password123
MasterPortDestroyer is armed
UltraGrid 1.5 (rev 1846029 built Feb 23 2019 07:32:15)

Display device   : sdl
Capture device   : none
Audio capture    : none
Audio playback   : none
MTU              : 1500 B
Video compression: none
Audio codec      : PCM
Network protocol : UltraGrid RTP
Audio FEC        : none
Video FEC        : none

[SDL] Using driver: cocoa
SDL2 initialized successfully.
Display initialized-sdl
Video capture initialized-none
Created new RTP session with SSRC 0x301afa0d.
[SDL] Reconfigure to size 512x512
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
[SDL] Using renderer: opengl
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Receiving unencrypted video data while expecting encrypted.
Video dec stats (cumulative): 0 total / 0 disp / 0 drop / 0 corr / 0 missing.
Pbuf: total 122444/122500 packets received (99.95429%).
```

**Comment by MartinPulec** [comment]
I confirm the issue - it turned out that combination of encryption+Reed-Solomon is unimplemented (LDGM implemented is). It may be quite trivial to implement, though.

**Comment by alatteri** [comment]
Thanks for the clarification.  My interest was in RS as the documentation states that LDGM is not useful with H.264.

**Comment by MartinPulec** [comment]
This should be fixed in master with commit 5ce362f2.
